test/rules/no-named-as-default.spec.ts

```diff
@@ -133,5 +137,16 @@ ruleTester.run('no-named-as-default', rule, {
         parserOptions: { ecmaVersion: 2022 },
       },
     }),
+
+    test({
+      code: `import z from 'zod';`,
+      errors: [
+        {
+          message:
+            "Using exported name 'z' as identifier for default export.",
+          type: 'ImportDefaultSpecifier',
+        },
+      ],
+    })
```

Comment:
@controversial This failed because you can do both:

```ts
import z from 'zod';
import { z } from 'zod';
```

And the rule `no-named-as-default` prefers the later.